src/journeys_utils.js

```diff
@@ -908,3 +908,72 @@ journeys_utils.setJourneyLinkData = function(linkData) {
 	}
 	journeys_utils.journeyLinkData = data;
 };
+
+journeys_utils.exitsValueInBranchViewData = function(name) {
+	if(!journeys_utils){
+		return false;
+	}
+
+	if(!journeys_utils.branch){
+		return false;
+	}
+
+	if(!journeys_utils.branch._branchViewData){
+		return false;
+	}
+
+	if(!journeys_utils.branch._branchViewData.data){
+		return false;
+	}
+
+	return journeys_utils.branch._branchViewData.data[name];
+};
+
+journeys_utils.hasJourneyCtaLink = function () {
+	if(!journeys_utils.exitsValueInBranchViewData('$journeys_cta')){
+		return false;
+	}
+
+	return journeys_utils.getBranchViewDataItemOrUndefined('$journeys_cta').length > 0;
+};
+
+journeys_utils.getBranchViewDataItemOrUndefined = function(name){
+	if(journeys_utils.exitsValueInBranchViewData(name)){
+		return journeys_utils.branch._branchViewData.data[name];
+	}
+	 return undefined;
+};
+
+journeys_utils.getJourneyCtaLink = function () {
+	return journeys_utils.getBranchViewDataItemOrUndefined('$journeys_cta');
+};
+
+journeys_utils.trySetBranchViewDataUrls = function (linkElements, urls = ['$android_url', '$ios_url', '$fallback_url', '$desktop_url']) {
+	if(!linkElements){
+		console.log("setDefaultUrls - no data set");
+		return linkElements;
+	}
+
+	var assignUrls = function (data) {
+		return urls.reduce((value, url)=>{
+			if(value[url]){
+				return value;
+			}
+
+			var entry = journeys_utils.getBranchViewDataItemOrUndefined(url);
+			if(entry){
+				value[url] = entry;
+			}
+			return value;
+		}, data);
+	};
+
+	try {
+		var data = (JSON.parse(linkElements.data));
+		linkElements.data = JSON.stringify(assignUrls(data));
+	}catch(e){
+		console.log(e);
```

Comment:
Are we sure we want to print errors to the console?

Reply:
In some way, we have to notify someone if there is an exception. We don't have a logging system for the SDK so only console log works. But I have an idea to deal with that. I'll try something 👍